### PR TITLE
Auto-generate Solana certs

### LINKED ISSUE #450: quic: generate cert for testing
**Problem**

The QUIC tests cannot run in CI yet because they are looking for `cert.pem` and `key.pem` files.

**Suggested Fix**

Use the OpenSSL C API to generate a certificate on the fly for certain unit tests.

### PR DESCRIPTION
**Pull request by ripatel-fd**
- ed25519: add OpenSSL EVP_PKEY wrap function
- x509: add Solana cert generator
- quic: use auto-generated cert
- quic: fix memory leaks in tests
- contrib: remove gen-pem-cert.sh
- quic: add tests to CI
- deps: OpenSSL 3.0.9-quic1
- fdctl: remove certs stage

Fixes #450

### CODE REVIEW

File: contrib/gen-pem-cert.sh [hunk 1]
```diff
@@ -1,10 +0,0 @@
-#!/bin/sh
```
Comment (mmcgee-jump) [inline contrib/gen-pem-cert.sh:1]: Think you need to remove the call to this in fd_frank_init, and remove `certs.rs` as well.

Comment (mmcgee-jump) [reply contrib/gen-pem-cert.sh:1]: (And remove the args provided to `fd_frank_run` ?)

Comment (mmcgee-jump) [reply contrib/gen-pem-cert.sh:1]: Also update `default.toml`, the comment about the `scratch_directory` storing certificates is no longer correct

File: src/tango/quic/fd_quic.c [hunk 2]
```diff
@@ -327,13 +320,34 @@ fd_quic_init( fd_quic_t * quic ) {
...
-  if( FD_UNLIKELY( !config->idle_timeout ) ) { FD_LOG_WARNING(( "zero cfg.idle_timeout" )); return NULL; }
+  if( FD_UNLIKELY( !config->role          ) ) { FD_LOG_WARNING(( "cfg.role not set"      )); return NULL; }
+  if( FD_UNLIKELY( !config->idle_timeout  ) ) { FD_LOG_WARNING(( "zero cfg.idle_timeout" )); return NULL; }
+
+  if( FD_UNLIKELY( (!quic->cert_object) | (!quic->cert_key_object) ) ) {
+    /* FIXME remove this hack by separating TLS and QUIC management.
+             User should provide pre-initialized SSL_CTX with cert
+             installed.  fd_quic should not touch any certificate
+             handling. */
+    FD_LOG_WARNING(( "Warning: Certificate or key not set. Generating random." ));
```
Comment (mmcgee-jump) [inline src/tango/quic/fd_quic.c:331]: Is this a warning? I thought we will run this way normally?

**Review by mmcgee-jump — DISMISSED** [review]

File: deps.sh [hunk 3]
```diff
@@ -108,7 +108,7 @@ fetch () {
   checkout_repo zlib      https://github.com/madler/zlib            "v1.2.13"
   checkout_repo bzip2     https://sourceware.org/git/bzip2.git      "bzip2-1.0.8"
   checkout_repo zstd      https://github.com/facebook/zstd          "v1.5.4"
-  checkout_repo openssl   https://github.com/quictls/openssl        "OpenSSL_1_1_1t-quic1"
+  checkout_repo openssl   https://github.com/quictls/openssl        "openssl-3.0.9-quic1"
```
Comment (lidatong) [inline deps.sh:111]: this is a backwards compatible upgrade right?

File: src/ballet/x509/fd_x509.c [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    FD_LOG_WARNING(( "X509_new() failed" ));
+    goto cleanup0;
+  }
+
+  X509_set_version( x, 2 );
+
+  /* Generate serial number */
+  long serial;
+  if( FD_UNLIKELY( 1!=RAND_bytes( (uchar *)&serial, sizeof(long) ) ) ) {
+    FD_LOG_WARNING(( "RAND_bytes() failed" ));
```
Comment (lidatong) [inline src/ballet/x509/fd_x509.c:58]: if something like cryptographically secure rng is failing should it just ERR and halt instead?

File: src/ballet/x509/fd_x509.h [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+#include "../fd_ballet_base.h"
+
+#if FD_HAS_OPENSSL
+
+#include <openssl/x509.h>
+
+/* FD_X509_SOLANA_CERT_SZ is the max size of a Solana cert. */
+
+#define FD_X509_SOLANA_CERT_SZ (512UL)
```
Comment (lidatong) [inline src/ballet/x509/fd_x509.h:14]: where does 512 come from?

File: src/tango/quic/fd_quic.c [hunk 6]
```diff
@@ -327,13 +320,34 @@ fd_quic_init( fd_quic_t * quic ) {
...
+    if( FD_UNLIKELY( quic->cert_object     ) ) X509_free    ( (X509 *)quic->cert_object         );
+    if( FD_UNLIKELY( quic->cert_key_object ) ) EVP_PKEY_free( (EVP_PKEY *)quic->cert_key_object );
+
+    /* Generate certificate key */
+    uchar cert_private_key[ 32 ];
+    FD_TEST( 1==RAND_bytes( cert_private_key, 32 ) );
```
Comment (lidatong) [inline src/tango/quic/fd_quic.c:337]: oh i should do this in my own code lol for some reason i only associated FD_TEST with tests

**Review by lidatong — APPROVED** [review]

**Review by lidatong — APPROVED** [review]

**Pull request merged**